larray_editor/editor.py

```diff
@@ -275,6 +279,29 @@ def get_value(self):
         # already been destroyed, due to the Qt.WA_DeleteOnClose attribute
         return self.data
 
+    def add_widget_with_state_to_restore(self, widget_name, widget):
+        if hasattr(widget, 'saveState'):
+            name = 'state/{}'.format(self.__class__.__name__)
+            if widget_name:
+                name += '_{}'.format(widget_name)
+                self.widget_state_settings[name] = widget
+
+    def closeEvent(self, event):
+        settings = QSettings()
+        settings.setValue('geometry/{}'.format(self.__class__.__name__), self.saveGeometry())
+        for setting_name, widget in self.widget_state_settings.items():
+            settings.setValue(setting_name, widget.saveState())
+
+    def showEvent(self, event):
```

Comment:
I don't know when showEvent is called but I hope it is not also called when unminimizing a window or when activating a window after it lost focus to another window, because that would make it annoying if you resize a window, focus on another window and come back (it would resize the window to its last saved state).